packages/graph-retriever/src/graph_retriever/adapters/base.py

```diff
@@ -376,10 +377,11 @@ def _metadata_filter(
         :
             The metadata dictionary to use for the given filter.
         """
-        metadata_filter = {**(base_filter or {})}
         assert isinstance(edge, MetadataEdge)
-        if edge is None:
-            metadata_filter
-        else:
-            metadata_filter[edge.incoming_field] = edge.value
-        return metadata_filter
+        value = edge.value
```

Comment:
Mostly some simplification (since `isinstance(edge, MetdataEdge)` implies it is not `None`).

Did take care to convert `immutabledict` back to `dict` so that downstream libraries don't get surprises if they do `isinstance(value, dict)`.